src/Compilers/Core/Portable/InternalUtilities/VoidResult.cs

```diff
@@ -2,10 +2,22 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System;
+
 namespace Roslyn.Utilities
 {
     /// <summary>
     /// Explicitly indicates result is void
     /// </summary>
-    internal readonly struct VoidResult { }
+    internal readonly struct VoidResult : IEquatable<VoidResult>
```

Comment:
💭 Interesting. Are these used anywhere it matters?

Reply:
we have an (optional) set we keep in the the batching queue to dedupe items if the client wants that.  i wanted to make sure that HashSet<VoidResult> would nto be a problem in that regard.